app/src/main/java/pl/example/android/garbageapp/ui/BaseActivitySector.java

```diff
@@ -82,16 +83,18 @@ protected boolean isMarkedForNotification() {
         return notificationSectorColor == SectorColor.toInt(sectorColor());
     }
 
-    protected void markForNotification(View view) {
+    protected void markForNotification(CompoundButton buttonView, boolean isChecked) {
         SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(currentSector());
         SharedPreferences.Editor editor = sharedPreferences.edit();
-        editor.putInt(NotificationUtils.NOTIFICATION_SECTOR_COLOR, SectorColor.toInt(sectorColor()));
-        editor.apply();
-        if (((Switch) view).isChecked())
+
+        if (isChecked) {
+            editor.putInt(NotificationUtils.NOTIFICATION_SECTOR_COLOR, SectorColor.toInt(sectorColor()));
```

Comment:
Extract this editor.putInt(...) into separate method. Pass sectorColor() as parameter. Use here and in else statement.

Reply:
Changed